src/index.js

```diff
@@ -4,62 +4,84 @@ import NavBar from './nav-bar.js'
 import ViewCards from './view-cards.js'
 import CardForm from './card-form.js'
 import PracticeCards from './practice-cards.js'
+import uuid from 'uuid/v4'
 
 class App extends React.Component {
   constructor(props) {
     super(props)
-    const initialCards = JSON.parse(localStorage.getItem('flashcards')) || []
-    const initialId = JSON.parse(localStorage.getItem('nextId')) || 1
     this.state = {
       view: 'view',
-      flashcards: initialCards,
-      selectedCard: null,
-      nextId: initialId
+      flashcards: [],
+      selectedCard: null
     }
-    this.onUnload = this.onUnload.bind(this)
     this.updateView = this.updateView.bind(this)
     this.addCard = this.addCard.bind(this)
     this.deleteCard = this.deleteCard.bind(this)
     this.updateSelected = this.updateSelected.bind(this)
   }
 
   componentDidMount() {
-    window.addEventListener('beforeunload', this.onUnload)
-  }
-
-  onUnload() {
-    localStorage.setItem('flashcards', JSON.stringify(this.state.flashcards))
-    localStorage.setItem('nextId', JSON.stringify(this.state.nextId))
+    fetch('/cards')
+      .then(res => res.json())
+      .then(cards => this.setState({ flashcards: cards }))
+      .catch(err => console.log(err))
   }
 
   addCard(card) {
     const currentCards = [...this.state.flashcards]
     if (this.state.selectedCard === null) {
-      const id = this.state.nextId
-      currentCards.push(Object.assign({}, card, { id: id }))
-      this.setState({
-        nextId: id + 1
-      })
+      const req = {
+        method: 'POST',
+        body: JSON.stringify(Object.assign({}, card, { id: uuid() })),
+        headers: { 'Content-Type': 'application/json' }
+      }
+      fetch('/cards', req)
+        .then(res => res.json())
+        .then(cardToAdd => {
+          currentCards.push(cardToAdd)
+          this.setState({
+            flashcards: currentCards,
+            selectedCard: null
+          })
+        })
+        .catch(err => console.log(err))
     }
     else {
-      const cardToUpdate = currentCards.find(card => card.id === this.state.selectedCard)
-      cardToUpdate.topic = card.topic
-      cardToUpdate.sideA = card.sideA
-      cardToUpdate.sideB = card.sideB
+      let cardToUpdate = currentCards.find(card => card.id === this.state.selectedCard)
+      const req = {
```

Comment:
See if you can remove some of the duplication here between your `POST` and `PUT` requests.